feature-libs/product/multi-dimensional/core/services/product-multi-dimensional.service.ts

```diff
@@ -6,66 +6,67 @@
 
 import { inject, Injectable } from '@angular/core';
 import { Product, VariantMatrixElement } from '@spartacus/core';
-import { VariantCategory } from '../model/augmented-core.model';
 import { ProductMultiDimensionalImagesService } from './product-multi-dimensional-images.service';
+import {
+  VariantCategoryGroup,
+  VariantCategoryOption,
+} from '@spartacus/product/multi-dimensional/root';
 
 @Injectable({ providedIn: 'root' })
 export class ProductMultiDimensionalService {
   protected imagesService = inject(ProductMultiDimensionalImagesService);
 
-  getVariants(product: Product): VariantCategory[] {
+  /**
+   * Retrieves variant categories for a given product, extracting options recursively
+   * from the variant matrix and adjusting for currently selected elements.
+   *
+   * @param {Product} product - The product object containing variant data.
+   * @returns {VariantCategoryGroup[]} - An array of variant categories with their options.
+   *
+   * @description
+   * This method processes a product's variant matrix to create an array of variant categories,
+   * each containing its respective options. It handles model conversions to ensure that the
+   * returned data is structured appropriately for the UI. Without these conversions, the UI
+   * would not be able to display the variant options correctly, and some attributes required
+   * for navigation would be missing.
+   *
+   * @example
+   * Given initially selected option "Blue 2 L":
+   * Color:        Red             (Blue)
+   * Size:      1     2         1      (2)
+   * Fit:     M  L  M   L     M   L   M   (L)
+   *
+   * The method extracts all available colors (e.g., Red and Blue) and recursively
+   * retrieves options until all variant options are exhausted.
+   *
+   * @see checkIfEveryOptionHasImages
+   * @see imagesService.getVariantOptionImages
+   */
+  getVariants(product: Product): VariantCategoryGroup[] {
```

Comment:
I've added more details:
 ```
 The method extracts all available colors (e.g., Red and Blue) and recursively retrieves options
until all variant options are exhausted, based on the currently selected options. For example the second row
contains multiple sizes of the same value, representing different products. Since the user can change
only one variant at a time, we need to identify only the products that reflect the current selections,
as shown in the tree with rows like (Blue), (2), (L).

Color:        Red             (Blue)
Size:                       1      (2)
Fit:                              M   (L)
 ```
 Question for bigger context:
For the order grid, a different method will be used to filter out values based on different assumptions. For example, the grid might display:

- Columns -> Sizes

- Rows -> Colors

I don't know many details about it, but I know that the exact display of the order grid values has not been finalized and is still to be decided.